NitroxClient/GameLogic/Spawning/Metadata/CyclopsMetadataProcessor.cs

```diff
@@ -0,0 +1,141 @@
+        }
+
+        if (Player.main.currentSub != engineState.subRoot)
+        {
+            engineState.startEngine = !isOn;
+            engineState.invalidButton = true;
+            engineState.Invoke(nameof(CyclopsEngineChangeState.ResetInvalidButton), 2.5f);
+            engineState.subRoot.BroadcastMessage("InvokeChangeEngineState", !isOn, SendMessageOptions.RequireReceiver);
+        }
+        else
+        {
+            engineState.invalidButton = false;
+            using (packetSender.Suppress<EntityMetadataUpdate>())
+            {
+                engineState.OnClick();
+            }
+        }        
+    }
+
+    private void SetEngineMode(GameObject cyclops, CyclopsMotorMode.CyclopsMotorModes mode)
+    {
+        foreach (CyclopsMotorModeButton button in cyclops.GetComponentsInChildren<CyclopsMotorModeButton>(true))
+        {
+            // At initial sync, this kind of processor is executed before the Start()
+            if (button.subRoot == null)
+            {
+                button.Start();
+            }
+
+            button.SetCyclopsMotorMode(mode);
+        }
+    }
+
+    private void ChangeSilentRunning(GameObject cyclops, bool isOn)
+    {
+        CyclopsSilentRunningAbilityButton ability = cyclops.RequireComponentInChildren<CyclopsSilentRunningAbilityButton>(true);
+
+        if (isOn == ability.active)
+        {
+            return;
+        }
+
+        Log.Debug($"Set silent running to {isOn} for cyclops");
+        ability.active = isOn;
+        if (isOn)
+        {
+            ability.image.sprite = ability.activeSprite;
+            ability.subRoot.BroadcastMessage("RigForSilentRunning");
+            ability.InvokeRepeating(nameof(CyclopsSilentRunningAbilityButton.SilentRunningIteration), 0f, ability.silentRunningIteration);
+        }
+        else
+        {
+            ability.image.sprite = ability.inactiveSprite;
+            ability.subRoot.BroadcastMessage("SecureFromSilentRunning");
+            ability.CancelInvoke(nameof(CyclopsSilentRunningAbilityButton.SilentRunningIteration));
+        }
+    }
+
+    private void ChangeShieldMode(GameObject cyclops, bool isOn)
+    {
+        CyclopsShieldButton shield = cyclops.GetComponentInChildren<CyclopsShieldButton>(true);
+
+        if (!shield)
+        {
+            // may not have a shield installed.
+            return;
+        }
+
+        bool isShieldOn = (shield.activeSprite == shield.image.sprite);
+
+        if (isShieldOn == isOn)
+        {
+            return;
+        }
+
+        if (isOn)
+        {
+            shield.StartShield();
+        }
+        else
+        {
+            shield.StopShield();
+        } 
+    }
+
+    private void ChangeSonarMode(GameObject cyclops, bool isOn)
+    {
+        CyclopsSonarButton sonarButton = cyclops.GetComponentInChildren<CyclopsSonarButton>(true);
+
+        if (sonarButton)
+        {
+            sonarButton.sonarActive = sonarButton._sonarActive;            
```

Comment:
```suggestion
            sonarButton.sonarActive = isOn;
```
Am pretty sure this is a mistake